### PR TITLE
gh-111968: Explicit handling for finalized freelist

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-111968

### CODE REVIEW

**Comment by corona10** [comment]
Since we marked as finalized at runtime, it should be done.
see: https://github.com/python/cpython/pull/113886#discussion_r1447498656 and https://github.com/python/cpython/pull/113921#discussion_r1447680872

File: Objects/floatobject.c [hunk 1]
```diff
@@ -131,11 +131,7 @@ PyFloat_FromDouble(double fval)
 #ifdef WITH_FREELISTS
     struct _Py_float_state *state = get_float_state();
     op = state->free_list;
-    if (op != NULL) {
-#ifdef Py_DEBUG
-        // PyFloat_FromDouble() must not be called after _PyFloat_Fini()
-        assert(state->numfree != -1);
-#endif
+    if (op != NULL && state->numfree > 0) {
```
Comment (colesbury) [inline Objects/floatobject.c]: I don't think we need extra checks when allocating from the free-lists. The existing checks should be sufficient.

**Review by colesbury — APPROVED** [review]
The check in `PyFloat_FromDouble` does not seem necessary, but it's up to you.

File: Objects/floatobject.c (hunk 1 above)
Comment (corona10) [reply Objects/floatobject.c]: Yeah you are correct!

**Pull request merged**